scripts/imgtool/image.py

```diff
@@ -139,7 +148,7 @@ def sign(self, key):
         sha.update(self.payload)
         digest = sha.digest()
 
-        tlv.add('SHA256', digest)
+        tlv.add('SHA256', digest, self.endian)
```

Comment:
I suggest you add endianness to the TLV constructor, so there's not need to supply it on `add` and `get`.